### PR TITLE
Webadmin Integration Test Junit 5 migration

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

File: server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQJwtFilterIntegrationTest.java [hunk 1]
```diff
@@ -19,27 +19,37 @@
...
+        .extension(new DockerElasticSearchExtension())
+        .extension(new CassandraExtension())
+        .extension(new AwsS3BlobStoreExtension())
+        .extension(new RabbitMQExtension())
+        .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
+            .combineWith(CassandraRabbitMQJamesServerMain.MODULES)
+            .overrideWith(new TestJMAPServerModule(LIMIT_TO_10_MESSAGES))
+            .overrideWith(binder -> binder.bind(AuthenticationFilter.class).to(JwtFilter.class))
+            .overrideWith(binder -> binder.bind(JwtConfiguration.class).toInstance(jwtConfiguration()))
+            .overrideWith(binder -> binder.bind(WebAdminConfiguration.class).toInstance(WebAdminConfiguration.TEST_CONFIGURATION)))
```
Comment (chibenwa) [inline server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQJwtFilterIntegrationTest.java]:
```
            .overrideWith(binder -> binder.bind(AuthenticationFilter.class).to(JwtFilter.class))
            .overrideWith(binder -> binder.bind(JwtConfiguration.class).toInstance(jwtConfiguration()))
            .overrideWith(binder -> binder.bind(WebAdminConfiguration.class).toInstance(WebAdminConfiguration.TEST_CONFIGURATION)))
```

Can't we have an `WebAdminIntegrationTestModule` ?

File: server/protocols/webadmin-integration-test/memory-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/memory/MemoryForwardIntegrationTest.java [hunk 2]
```diff
@@ -19,22 +19,24 @@
...
 import org.apache.james.webadmin.integration.ForwardIntegrationTest;
-import org.junit.Rule;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
-public class MemoryForwardIntegrationTest extends ForwardIntegrationTest {
+class MemoryForwardIntegrationTest extends ForwardIntegrationTest {
 
-    @Rule
-    public MemoryJmapTestRule memoryJmap = new MemoryJmapTestRule();
+    private static final int LIMIT_TO_10_MESSAGES = 10;
```
Comment (Arsnael) [inline server/protocols/webadmin-integration-test/memory-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/memory/MemoryForwardIntegrationTest.java]: As stated and agreed with @mbaechler before when I was migrating test classes, JUnit5 is allowing us to reduce verbosity of the code, by not declaring necessarily the access modifiers (which we were doing already for classes and  test method declarations). We tend to remove that with global variables too now.

**Comment by trantienduchn** [comment]
Build time on other builds:
```
[INFO] Apache James :: Server :: Web Admin server integration tests :: Distributed SUCCESS [04:03 min]
[INFO] Apache James :: Server :: Web Admin server integration tests :: Memory SUCCESS [03:09 min]

```

File: server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQJwtFilterIntegrationTest.java (hunk 1 above)
Comment (trantienduchn) [reply server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQJwtFilterIntegrationTest.java]: Miss reading, I created a new class instead. But I think it's acceptable

**Comment by trantienduchn** [comment]
:green_apple:

File: server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/vault/RabbitMQDeletedMessageVaultIntegrationTest.java [hunk 3]
```diff
@@ -42,7 +42,7 @@
     public GuiceJamesServer createJmapServer(FileSystem fileSystem, Clock clock) throws IOException {
         return rule.jmapServer(cassandra.getModule(),
```
Comment (trantienduchn) [inline server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/vault/RabbitMQDeletedMessageVaultIntegrationTest.java]: `DeletedMessageVaultIntegrationTest` is a special case, I will do it in a separated PR

File: server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQEventDeadLettersIntegrationTest.java [hunk 4]
```diff
@@ -207,8 +232,8 @@ private String retrieveFirstFailedInsertionId() {
     }
 
     @Test
-    public void failedEventShouldBeStoredInDeadLetterUnderItsGroupId() {
-        retryEventsListener.callsPerEventBeforeSuccess(MAX_RETRIES + 1);
+    void failedEventShouldBeStoredInDeadLetterUnderItsGroupId() {
+        EVENT_EXTENSION.retryEventsListener().callsPerEventBeforeSuccess(MAX_RETRIES + 1);
```
Comment (chibenwa) [inline server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQEventDeadLettersIntegrationTest.java]: This, looks like evil to me.

A more idiomatic way to do this (that also comply to visibility best practices) is to use `@Register` on eventExtension, then make it resolve parameter of type `RetryEventsListener`, and finally pass a parameter to the test classes?

File: server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/vault/RabbitMQDeletedMessageVaultIntegrationTest.java (hunk 3 above)
Comment (trantienduchn) [reply server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/vault/RabbitMQDeletedMessageVaultIntegrationTest.java]: not any more

File: server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/WebadminIntergrationTestModule.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.integration;
+
+import org.apache.james.webadmin.WebAdminConfiguration;
+
+import com.google.inject.AbstractModule;
+
+public class WebadminIntergrationTestModule extends AbstractModule {
```
Comment (Arsnael) [inline server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/WebadminIntergrationTestModule.java]: WebAdminIntegrationTestModule

File: server/protocols/webadmin-integration-test/memory-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/memory/MemoryForwardIntegrationTest.java (hunk 2 above)
Comment (Arsnael) [reply server/protocols/webadmin-integration-test/memory-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/memory/MemoryForwardIntegrationTest.java]: Didn't see any changes on this, why resolved? :)

Comment (trantienduchn) [reply server/protocols/webadmin-integration-test/memory-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/memory/MemoryForwardIntegrationTest.java]: I missed understand you, but I had remove all `LIMIT_TO_10_MESSAGES` declarations to use an unique constant in `TestJMAPServerModule`. That constant could be accessed from many places in the code base, so I would like to keep it public as its scope now not only inside the class.

**Comment by trantienduchn** [comment]
```
[580e239fae40998be38ec3f046e720c6bf9aa577] [ERROR] Errors: 
[580e239fae40998be38ec3f046e720c6bf9aa577] [ERROR]   JamesDeploymentValidationTest>DeploymentValidation.validateDeploymentWithMailsFromSmtp:95 » ConditionTimeout
[580e239fae40998be38ec3f046e720c6bf9aa577] [INFO] 
[580e239fae40998be38ec3f046e720c6bf9aa577] [ERROR] Tests run: 5, Failures: 0, Errors: 1, Skipped: 2
```

test this please

**Comment by trantienduchn** [comment]
squashed to welcome cross reviewers

File: server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQWebAdminServerIntegrationTest.java [hunk 6]
```diff
@@ -41,32 +48,36 @@
...
 import org.eclipse.jetty.http.HttpStatus;
-import org.junit.Rule;
-import org.junit.Test;
 import org.junit.experimental.categories.Category;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
 import io.restassured.http.ContentType;
 
 @Category(BasicFeature.class)
```
Comment (mbaechler) [inline server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQWebAdminServerIntegrationTest.java]: does @Category work the same way with junit5?

File: server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQWebAdminServerIntegrationTest.java [hunk 7]
```diff
@@ -41,32 +48,36 @@
...
+
+    @RegisterExtension
+    static JamesServerExtension testExtension = new JamesServerBuilder()
+        .extension(new DockerElasticSearchExtension())
+        .extension(new CassandraExtension())
+        .extension(new AwsS3BlobStoreExtension())
+        .extension(new RabbitMQExtension())
+        .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
+            .combineWith(CassandraRabbitMQJamesServerMain.MODULES)
+            .overrideWith(new TestJMAPServerModule(LIMIT_TO_10_MESSAGES))
```
Comment (mbaechler) [inline server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQWebAdminServerIntegrationTest.java]: I don't think this testsuite does any JMAP request. If not, you could remove this override.

File: server/protocols/webadmin-integration-test/memory-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/memory/MemoryWebAdminServerIntegrationTest.java [hunk 8]
```diff
@@ -20,19 +20,23 @@
...
-    @Override
-    protected GuiceJamesServer createJamesServer() throws Exception {
-        return memoryJmap.jmapServer(binder -> binder.bind(WebAdminConfiguration.class)
-            .toInstance(WebAdminConfiguration.TEST_CONFIGURATION));
-    }
+    @RegisterExtension
+    static JamesServerExtension jamesServerExtension = new JamesServerBuilder()
+        .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
+            .combineWith(MemoryJamesServerMain.IN_MEMORY_SERVER_AGGREGATE_MODULE)
+            .overrideWith(new TestJMAPServerModule(LIMIT_TO_10_MESSAGES))
```
Comment (mbaechler) [inline server/protocols/webadmin-integration-test/memory-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/memory/MemoryWebAdminServerIntegrationTest.java]: same here

File: server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java [hunk 9]
```diff
@@ -69,13 +68,10 @@
...
 
-    private GuiceJamesServer guiceJamesServer;
     private DataProbe dataProbe;
 
-    @Before
-    public void setUp() throws Exception {
-        guiceJamesServer = createJamesServer();
-        guiceJamesServer.start();
+    @BeforeEach
+    void setUp(GuiceJamesServer guiceJamesServer) throws Exception {
```
Comment (mbaechler) [inline server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java]: why don't you make it a contract test (interface)?

File: server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQJwtFilterIntegrationTest.java [hunk 10]
```diff
@@ -19,27 +19,37 @@
...
-    }
+    @RegisterExtension
+    static JamesServerExtension testExtension = new JamesServerBuilder()
+        .extension(new DockerElasticSearchExtension())
+        .extension(new CassandraExtension())
+        .extension(new AwsS3BlobStoreExtension())
+        .extension(new RabbitMQExtension())
+        .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
+            .combineWith(CassandraRabbitMQJamesServerMain.MODULES)
+            .overrideWith(new TestJMAPServerModule(LIMIT_TO_10_MESSAGES))
```
Comment (mbaechler) [inline server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQJwtFilterIntegrationTest.java]: same here

File: server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/JwtFilterIntegrationTest.java [hunk 11]
```diff
@@ -34,15 +34,19 @@
...
-import org.junit.After;
-import org.junit.Before;
-import org.junit.Test;
 import org.junit.experimental.categories.Category;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
 
 import io.restassured.RestAssured;
 
 public abstract class JwtFilterIntegrationTest {
```
Comment (mbaechler) [inline server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/JwtFilterIntegrationTest.java]: why not a interface?

File: server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/FixingGhostMailboxTest.java [hunk 12]
```diff
@@ -126,7 +126,7 @@
     private RequestSpecification webadminSpecification;
 
     @BeforeEach
-    void setup(GuiceJamesServer server) throws Throwable {
+    void setup(GuiceJamesServer server, DockerCassandraRule cassandra) throws Throwable {
```
Comment (mbaechler) [inline server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/FixingGhostMailboxTest.java]: you don't want to inject the DockerCassandraRule, only the host, I guess.

File: server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQWebAdminServerIntegrationTest.java (hunk 6 above)
Comment (trantienduchn) [reply server/protocols/webadmin-integration-test/distributed-webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/rabbitmq/RabbitMQWebAdminServerIntegrationTest.java]: No, I cannot find a resource about it, but tested manually, surefire ignore junit 5 tests marked with `@Category` while it doesn't happen with `@Tag`

**Comment by trantienduchn** [comment]
```
[581305fbdab8ad4ad2f632972cc950e30fb55fa8] [ERROR] Failures: 
[581305fbdab8ad4ad2f632972cc950e30fb55fa8] [ERROR]   FixingGhostMailboxTest.ghostMailboxBugShouldDiscardOldContent:217 1 expectation failed.
```

File: server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java (hunk 9 above)
Comment (trantienduchn) [reply server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java]: Mostly I don't want to grow the change set. I will invest that effort in another PR if you don't mine

Comment (trantienduchn) [reply server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java]: see #3042

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**